gadget/install/install_test.go

```diff
@@ -66,6 +71,417 @@ func (s *installSuite) TestInstallRunError(c *C) {
 	c.Check(sys, IsNil)
 }
 
+func (s *installSuite) TestInstallRunSimpleHappy(c *C) {
+	s.testInstall(c, installOpts{
+		encryption: false,
+	})
+}
+
+func (s *installSuite) TestInstallRunEncryptedLUKS(c *C) {
+	s.testInstall(c, installOpts{
+		encryption: true,
+	})
+}
+
+func (s *installSuite) TestInstallRunExistingPartitions(c *C) {
+	s.testInstall(c, installOpts{
+		encryption:    false,
+		existingParts: true,
+	})
+}
+
+func (s *installSuite) TestInstallRunEncryptionExistingPartitions(c *C) {
+	s.testInstall(c, installOpts{
+		encryption:    true,
+		existingParts: true,
+	})
+}
+
+type installOpts struct {
+	encryption    bool
+	existingParts bool
+}
+
+func (s *installSuite) testInstall(c *C, opts installOpts) {
+	cleanups := []func(){}
+	addCleanup := func(r func()) { cleanups = append(cleanups, r) }
+	defer func() {
+		for _, r := range cleanups {
+			r()
+		}
+	}()
+
+	uc20Mod := &gadgettest.ModelCharacteristics{
+		SystemSeed: true,
+	}
+
+	s.setupMockUdevSymlinks(c, "mmcblk0p1")
+
+	// mock single partition mapping to a disk with only ubuntu-seed partition
+	initialDisk := gadgettest.ExpectedRaspiMockDiskInstallModeMapping
+	if opts.existingParts {
+		// unless we are asked to mock with a full existing disk
+		if opts.encryption {
+			initialDisk = gadgettest.ExpectedLUKSEncryptedRaspiMockDiskMapping
+		} else {
+			initialDisk = gadgettest.ExpectedRaspiMockDiskMapping
+		}
+	}
+	m := map[string]*disks.MockDiskMapping{
+		filepath.Join(s.dir, "/dev/mmcblk0p1"): initialDisk,
+	}
+
+	restore := disks.MockPartitionDeviceNodeToDiskMapping(m)
+	defer restore()
+
+	restore = disks.MockDeviceNameToDiskMapping(map[string]*disks.MockDiskMapping{
+		"/dev/mmcblk0": initialDisk,
+	})
+	defer restore()
+
+	mockSfdisk := testutil.MockCommand(c, "sfdisk", "")
```

Comment:
For extra completeness this could capture stdin to a file too

Reply:
Good point, it would be sort of nice if there was a MockCommand utility where we could check in Go what the stdin for the file was

Reply:
I think I will do this in a followup since this is green with 2 +1s